pkg/service/aws/cost.go

```diff
@@ -158,6 +158,160 @@ func (svc AWSController) GetWorkspacesCost(ctx context.Context, req *proto.GetWo
+					Key:    aws.String("LINKED_ACCOUNT"),
+					Values: aws.StringSlice([]string{*account_id}),
+				},
+			},
+			{
+				Tags: &costexplorer.TagValues{
+					Key:    aws.String(constants.WorkspaceId),
+					Values: aws.StringSlice(req.GetApplicationIds()),
+				},
+			},
+
+			{
+
+				Not: &costexplorer.Expression{
+					Dimensions: &costexplorer.DimensionValues{
+						Key: aws.String("RECORD_TYPE"),
+						Values: aws.StringSlice([]string{
+							"Credit",
+						}),
+					},
+				},
+			},
+		},
+	}
+
+	input := costexplorer.GetCostAndUsageInput{
+		TimePeriod: &costexplorer.DateInterval{
+			Start: aws.String(req.GetStartDate()),
+			End:   aws.String(req.EndDate),
+		},
+		Granularity: aws.String(req.GetGranularity()),
+		GroupBy: []*costexplorer.GroupDefinition{
+			{
+				Type: aws.String(req.GroupBy.Type),
+				Key:  aws.String(req.GroupBy.Key),
+			},
+		},
+		Metrics: aws.StringSlice([]string{req.GetCostType()}),
+		Filter:  filter,
+	}
+
+	return input
+}
+
+func getTotalCost(costMap map[string]decimal.Decimal, result *costexplorer.GetCostAndUsageOutput, costType string) (decimal.Decimal, error) {
+
+	var totalCost decimal.Decimal
+
+	for _, resultByTime := range result.ResultsByTime {
+
+		for _, group := range resultByTime.Groups {
+			groupKey := ""
+
+			for _, key := range group.Keys {
+
+				if key != nil {
+					key := strings.Split(*key, "$")[1]
+					groupKey += key
+				}
+			}
+
+			groupMetric, ok := group.Metrics[costType]
+			if !ok {
+				groupMetric = &costexplorer.MetricValue{
+					Amount: aws.String("0"),
+					Unit:   aws.String("USD"),
+				}
+			}
+
+			decimalCost, err := decimal.NewFromString(*groupMetric.Amount)
+			if err != nil {
+				return decimal.Zero, errors.Wrap(err, "GetWorkspacesCost: failed to convert amount to decimal")
+			}
+
+			costMap[groupKey] = costMap[groupKey].Add(decimalCost)
+			totalCost = totalCost.Add(decimalCost)
+
+		}
+
+	}
+
+	return totalCost, nil
+}
+func (svc AWSController) GetApplicationsCost(ctx context.Context, req *proto.GetApplicationsCostRequest) (*proto.GetApplicationsCostResponse, error) {
+
+	session, err := NewSession(ctx, "", req.GetAccountName())
+
+	if err != nil {
+		svc.logger.Errorw("can't start AWS session", "error", err)
+		return nil, err
+	}
+
+	stsClient := session.getSTSClient()
+
+	callerIdentity, err := stsClient.GetCallerIdentity(nil)
+
+	if err != nil {
+		svc.logger.Errorw("failed to get identity", "error", err)
+		return nil, err
+	}
+
+	account_id := callerIdentity.Account
+
+	svc.logger.Debugw("fetched accountId", "id", account_id)
+
+	input := getCostAndUsageRequest(account_id, req)
+	client := session.getCostExplorerClient()
+
+	result, err := client.GetCostAndUsage(&input)
+
+	if err != nil {
+		svc.logger.Errorw("failed to get cost ", "error", err)
+		return nil, err
+	}
+
+	costMap := make(map[string]decimal.Decimal)
+
+	var totalCost decimal.Decimal
+
+	totalCost, err = getTotalCost(costMap, result, req.GetCostType())
```

Comment:
rename this and get the costMap in response to this function only

Reply:
what should I rename this to?